lib/rust/ensogl/component/flame-graph/src/lib.rs

```diff
@@ -0,0 +1,298 @@
+//! An EnsoGL implementation of a basic flame graph.
+
+#![warn(missing_copy_implementations)]
+#![warn(missing_debug_implementations)]
+#![warn(missing_docs)]
+#![warn(trivial_casts)]
+#![warn(trivial_numeric_casts)]
+#![warn(unsafe_code)]
+#![warn(unused_import_braces)]
+#![warn(unused_qualifications)]
+
+
+use ensogl_core::prelude::*;
+
+use enso_frp as frp;
+use enso_profiler as profiler;
+use ensogl_core::application;
+use ensogl_core::application::shortcut;
+use ensogl_core::application::Application;
+use ensogl_core::data::color;
+use ensogl_core::display;
+use ensogl_core::display::scene::Layer;
+use ensogl_core::display::shape::*;
+use ensogl_core::Animation;
+use ensogl_text as text;
+
+
+
+// =======================
+// === Block Component ===
+// =======================
+
+/// A single block that makes up a flame graph.
+mod block {
```

Comment:
Why is this not a separate file? As a separate file it will have better indentation and will also make this file way shorter.